server/app/views/applicant/ApplicantProgramBlockEditTemplate.html

```diff
@@ -0,0 +1,12 @@
+<html>
+    <head th:replace="~{applicant/ApplicantBaseFragment :: pageHeader}"></head>
+    <body>
+        <div>
+            <form id="cf-block-form" th:action="${formAction}" method="POST">
+                <input hidden th:value="${csrfToken}" name="csrfToken">
+                <div th:replace="~{applicant/ApplicantBaseFragment :: requiredFieldsExplanation}"></div>
+            </form>
+        </div>
+        <footer th:replace="~{applicant/ApplicantBaseFragment :: pageFooter}"></footer>
```

Comment:
**question (thinking out loud, I don't actually have strong opinions here)**: With the pattern in this PR, there's a template for each page that then pulls in the header and footer fragments into itself from the `ApplicantBaseFragment`.

Why is that better then a pattern like this one?
https://www.thymeleaf.org/doc/tutorials/3.1/usingthymeleaf.html#layout-inheritance
in which there's an outer `layout` fragment that contains the header and footer, and the body content is inserted into it.

(It might be that it actually doesn't matter and we just picked the one that was simplest at first, I'm just curious if there were additional reasons.)

Reply:
For me, when I was making this PR at least, I think it comes down to readability, but I also don't a strongly held opinion. When I look at that example in the docs you link to, it doesn't feel like it would be immediately obvious to someone new to Thymeleaf what DOM structure that would generate. The obvious advantage of that pattern is that it's easier to reuse the more pages you have (although slightly less simple when there is variance among pages, like different page titles.)

I don't think this needs to be a permanent decision, it would be pretty easy to do that as a separate refactor after the fact as we add more pages.

Reply:
sgtm!